### ISSUE TITLE
False positive no-member for class attributes

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pylint#2425]
### Steps to reproduce

Run `pylint -E` against this code:

```python
class A:
    d = 'foo bar baz'
    d = set(d.split())

A.d.remove('foo')
```

### Current behavior

```
test.py:5:0: E1101: Instance of 'str' has no 'remove' member (no-member)
```

### Expected behavior

No complaints about missing member.

### pylint --version output

```
pylint 2.1.1
astroid 2.0.4
Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
[GCC 6.3.0 20170118]
```

**Comment by PCManticore** [comment]
Thank you, I can also reproduce this problem. This happens because we don't infer the last `d` assignment, stopping at the first one. It's also similar to https://github.com/PyCQA/pylint/issues/2336.

**Comment by craiglabenz** [comment]
Sounds hard to fix, but fwiw, this is also triggers the false-positive:

```py
class A:
    def __init__(self):
        val = 'b'
        setattr(self, val, True)

a = A()
print(a.b)  # E1101:Instance of 'A' has no 'b' member
```

**Comment by PCManticore** [comment]
@craiglabenz :wave: That seems a separate problem. `pylint` cannot correctly understand `setattr`, as it mutates objects at runtime. While we do have limited support for this kind of dynamic constructs (e.g. `getattr`) there's nothing that could help us infer `setattr` correctly. For your use case, passing `--generated-members=b` or `--ignored-classes=A` should do it.
